src/test/java/de/retest/descriptors/IdentifyingAttributesTest.java

```diff
@@ -0,0 +1,159 @@
+	}
+
+	@Test
+	public void no_out_of_two_attributes_should_match_0_percent() {
+		final IdentifyingAttributes expected =
+				IdentifyingAttributes.create( Path.fromString( "a/component_X" ), component.class );
+		final IdentifyingAttributes actual =
+				IdentifyingAttributes.create( Path.fromString( "b/component_Y" ), otherComponent.class );
+
+		assertThat( expected ).isNotEqualTo( actual );
+		assertThat( expected.hashCode() ).isNotEqualTo( actual.hashCode() );
+		assertThat( expected.match( actual ) ).isCloseTo( 0.0, within( 0.01 ) );
+	}
+
+	@Test
+	public void only_same_parent_path_results_in_1_similarity() {
+		// TODO Maybe the counter should be indifferent to type?
+		final IdentifyingAttributes expected =
+				IdentifyingAttributes.create( Path.fromString( "Window/path/component_X" ), component.class );
+		final IdentifyingAttributes actual =
+				IdentifyingAttributes.create( Path.fromString( "Window/path/component_Y" ), otherComponent.class );
+
+		assertThat( expected.match( actual ) ).isCloseTo( ONE_SIMILARITY, within( 0.01 ) );
+	}
+
+	@Test
+	public void several_differences_should_decrease_similarity() {
+		final IdentifyingAttributes cell_0_4_orig = IdentifyingAttributes.create( Path.fromString(
+				"Window/JRootPane_0/JLayeredPane_0/StatefulTableDemo_0/JTabbedPane_0/Tab_0/JPanel_0/JTable_0/row_0/column_4" ),
+				cell.class );
+		final IdentifyingAttributes cell_0_4_new = IdentifyingAttributes.create( Path.fromString(
+				"Window/JRootPane_0/JLayeredPane_0/StatefulTableDemo_0/JTabbedPane_0/Tab_1/JPanel_0/JTable_0/row_0/column_4" ),
+				cell.class );
+		final IdentifyingAttributes cell_2_4_new = IdentifyingAttributes.create( Path.fromString(
+				"Window/JRootPane_0/JLayeredPane_0/StatefulTableDemo_0/JTabbedPane_0/Tab_1/JPanel_0/JTable_0/row_2/column_4" ),
+				cell.class );
+
+		assertThat( cell_0_4_orig.match( cell_0_4_new ) ).isGreaterThan( cell_0_4_orig.match( cell_2_4_new ) );
+	}
+
+	@Test
+	public void apply_same_value_should_work() throws Exception {
+		final String pathValue = "Window/JRootPane_0/JLayeredPane_0/JTable_0/row_6/column_0";
+		final IdentifyingAttributes identifyingAttributes =
+				IdentifyingAttributes.create( fromString( pathValue ), component.class );
+		identifyingAttributes
+		.applyChanges( createAttributeChanges( fromString( pathValue ), "path", pathValue, pathValue ) );
+	}
+
+	@Test
+	public void apply_path_change() {
+		final IdentifyingAttributes identifyingAttributes = IdentifyingAttributes.create( path, component.class );
+
+		final AttributeDifference pathDifference = new AttributeDifference( "path",
+				Path.fromString( "Window/path/component_0" ), Path.fromString( "Window/otherPath/component_1" ) );
+		final AttributeDifference suffixDifference = new AttributeDifference( "suffix", "0", "1" );
+		final IdentifyingAttributes changed = identifyingAttributes
+				.applyChanges( new HashSet<>( Arrays.asList( pathDifference, suffixDifference ) ) );
```

Comment:
We moved from Java 8 to 6. Therefore, the diamond operator isn't feasible anymore and causes the current build to fail: https://travis-ci.com/retest/retest-model/builds/77030726.